WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController/NotificationsViewController.swift

```diff
@@ -381,6 +353,64 @@ class NotificationsViewController: UIViewController, UIViewControllerRestoration
         return .none
     }
 
+    private func cellContent(from notification: Notification, at indexPath: IndexPath) -> NotificationsTableViewCellContent {
+        // Handle the case of a deleted / spammed notification
+        if let deletionRequest = notificationDeletionRequests[notification.objectID] {
+            let style = NotificationsTableViewCellContent.Style.altered(
+                .init(
+                    text: deletionRequest.kind.legendText,
+                    action: { [weak self] in
+                        self?.cancelDeletionRequestForNoteWithID(notification.objectID)
+                    }
+                )
+            )
+            return .init(style: style)
+        }
+
+        // Handle the regular case
+        let title: AttributedString? = {
+            guard let attributedSubject = notification.renderSubject() else {
+                return nil
+            }
+            return AttributedString(attributedSubject)
+        }()
+        let description = notification.renderSnippet()?.string
+        let inlineAction = cellInlineAction(from: notification, at: indexPath)
+        let avatarStyle = AvatarsView.Style(urls: notification.allAvatarURLs) ?? .single(notification.iconURL)
+        let style = NotificationsTableViewCellContent.Style.regular(
+            .init(
+                title: title,
+                description: description,
+                shouldShowIndicator: !notification.read,
+                avatarStyle: avatarStyle,
+                inlineAction: inlineAction
+            )
+        )
+        return .init(style: style)
+    }
```

Comment:
No major changes here, I've simply extracted this logic from `cellForRowAtIndexPath` method into this function.